docs/en/observability/monitor-aws.asciidoc

```diff
@@ -0,0 +1,596 @@
+----
++
+[NOTE]
+=====
+Make sure you specify the `-n` parameter; otherwise, you will have
+painful debugging sessions due to adding a newline at the end of
+your API key.
+=====
+
+. To see if both settings have been stored, run the following command:
++
+[source,bash]
+----
+./metricbeat keystore list
+----
+
+. To configure {metricbeat} to output to {ess}, edit the `metricbeat.yml`
+configuration file. Add the following lines to the end of the file.
++
+[source,yml]
+----
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+----
+
+. Finally, test if the configuration is working. If it is not working,
+verify if you used the right credentials and add them again.
++
+[source,bash]
+----
+./metricbeat test output
+----
+
+Now that the output is working, you are going to set up the {aws} module.
+
+[discrete]
+==== Step 9: Configuring the {aws} module
+
+To collect the metrics from your {aws} infrastructure we will use the
+{metricbeat-ref}/metricbeat-module-aws.html[{aws}] module. The {aws} module is
+containing a lot of metricsets: `billing`, `cloudwatch`, `dynamodb`, `ebs`,
+`ec2`, `elb`, `lambda` and many more. Each metricsets is created to help you
+stream and process your data. In this tutorial we are going to show you a
+few examples using the `ec2` and the `billing` metricsets.
+
+. Let's enable the {aws} module in {metricbeat}.
++
+[source,bash]
+----
+./metricbeat modules enable aws
+----
+
+. Edit the `modules.d/aws.yml` file with the following configurations.
++
+[source,yml]
+----
+- module: aws <1>
+  period: 24h <2>
+  metricsets:
+    - billing <3>
+  credential_profile_name: mb-aws <4>
+  cost_explorer_config:
+    group_by_dimension_keys:
+      - "AZ"
+      - "INSTANCE_TYPE"
+      - "SERVICE"
+    group_by_tag_keys:
+      - "aws:createdBy"
+- module: aws <1>
+  period: 300s <2>
+  metricsets:
+    - ec2 <3>
+  credential_profile_name: mb-aws <4>
+----
++
+<1> Defines the module that is going to be used.
+<2> Defines the period at which the metrics are going to be collected
+<3> Defines the metricset that is going to be used.
+<4> This is the aws profile define following the https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html[AWS standard].
+
+Make sure that the {aws} user used to collect the metrics from CloudWatch has at
+least the following permissions attached to it:
+
+[source,yml]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "ec2:DescribeInstances",
+                "ec2:DescribeRegions",
+                "cloudwatch:GetMetricData",
+                "cloudwatch:ListMetrics",
+                "sts:GetCallerIdentity",
+                "iam:ListAccountAliases",
+                "tag:getResources",
+                "ce:GetCostAndUsage"
+            ],
+            "Effect": "Allow",
+            "Resource": "*"
+        }
+    ]
+}
+----
```

Comment:
Oh snap! Where is this for Filebeat? This is exactly what I needed above

Reply:
I found the following minimal permission:

```
s3:GetObject
sqs:ReceiveMessage
sqs:ChangeMessageVisibility
sqs:DeleteMessage
```